### PR TITLE
Add tracing_attributes kwargs to tracing policy

### PR DESCRIPTION
**Pull request by lmazuel**
Step in https://github.com/Azure/azure-sdk-for-python/issues/9256

@lmolkova

### CODE REVIEW

File: sdk/core/azure-core/azure/core/pipeline/policies/_distributed_tracing.py [hunk 1]
```diff
@@ -88,6 +89,8 @@ def on_request(self, request):
             span_name = namer(request.http_request)
 
             span = span_impl_type(name=span_name)
+            for attr, value in self._tracing_attributes.items():
```
Comment (johanste) [inline sdk/core/azure-core/azure/core/pipeline/policies/_distributed_tracing.py]: Are the `tracing_attributes` only supposed to be provided in `__init__`? (i.e. not on a per-request basis?)

Comment (lmazuel) [reply sdk/core/azure-core/azure/core/pipeline/policies/_distributed_tracing.py]: I think that's enough, but it's open design at this point :). What you feel?

File: sdk/core/azure-core/azure/core/pipeline/policies/_distributed_tracing.py [hunk 2]
```diff
@@ -75,6 +75,7 @@ class DistributedTracingPolicy(SansIOHTTPPolicy):
 
     def __init__(self, **kwargs):
         self._network_span_namer = kwargs.get('network_span_namer', _default_network_span_namer)
+        self._tracing_attributes = kwargs.get('tracing_attributes', {})
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/pipeline/policies/_distributed_tracing.py]: Please add docstring for "tracing_attributes"

File: sdk/core/azure-core/azure/core/pipeline/policies/_distributed_tracing.py (hunk 1 above)
Comment (johanste) [reply sdk/core/azure-core/azure/core/pipeline/policies/_distributed_tracing.py]: I guess the caller can always add attributes to the current (ambient) span directly if they had a need to do so (e.g. add a client request id or something similar). 

And if we do want to support the addition of per-request level parameters, it is an additive change.

File: sdk/core/azure-core/azure/core/pipeline/policies/_distributed_tracing.py (hunk 2 above)
Comment (lmazuel) [reply sdk/core/azure-core/azure/core/pipeline/policies/_distributed_tracing.py]: fixed

**Pull request merged**